stackrox/stackrox
# Delete obsolete form components from UI

## Description

Found while studying for meeting about Redux and sagas.

1. Delete from src/Components/forms folder:
    * FormFieldLabel/FormFieldLabel.js
    * FormFieldLabel/index.js
    * ReduxCheckboxField.js
    * ReduxColorPickerField.js
    * ReduxMultiSelectCreatableField.js
    * ReduxMultiSelectField.js
    * ReduxNumericInputField.js
    * ReduxRadioButtonGroupField.js
    * ReduxRadioField.js
    * ReduxSelectCreatableField.js
    * ReduxTextAreaField.js
    * ReduxToggleField.js
2. Move from src/Components/forms to src/Containers/Clusters/Components folder:
    * FormFieldRequired.js has 4 occurrences in StaticConfigurationSection.js
2. Delete from src/Components folder:
    * SaveButton.js

### Residue

After we replace redux-form with formik in `LoginPage` component:
1. Delete src/Components/forms folder which contains:
    * ReduxPasswordField.js
    * ReduxSelectField.js
    * ReduxTextField.js
2. Delete src/Components/Labeled subfolder which contains:
    * Labeled.js
    * index.js
3. Delete `form` reducer from Redux store.
4. Delete redux-form from `dependencies` for UI.

## Checklist

## Testing Performed

### manual testing

1. `yarn build` in ui
2. `wc build/static/js/*.chunk.js` in ui
    * branch - master: 0 = 5560522 - 5560522 confirms that the components were not referenced
3. `yarn start` in ui
4. Visit /main/clusters
    * Click a cluster: see **(required)** in gray
        ![clusters-required-view](https://user-images.githubusercontent.com/11862657/191524283-afcbac55-c0c0-4010-9972-29195173310d.png)

    * Start to create a new cluster: see **(required)** in yellow
        ![clusters-required-new](https://user-images.githubusercontent.com/11862657/191524316-8475e975-f242-4932-bc4f-832a4cbb6daa.png)
